real_climate.py

```diff
@@ -0,0 +1,284 @@
+                    # later_time is the closest time that is later than ms_time
+                    time_dif_ms = later_time['timestampMs'] - earlier_time['timestampMs']  # time difference in milliseconds 
+                    time_dif_min = self.ms_convert(time_dif_ms)   # ms time difference is converted to minutes 
+                    temp_dif = self.celsius_convert(later_time['temp']) - self.celsius_convert(earlier_time['temp'])   
+                    # temperature difference between the two times is calculated in Fahrenheit
+                    humid_dif = later_time['humidity'] - earlier_time['humidity']
+                    # humidity difference between the two times is calculated in Fahrenheit
+                    temp_rate_min = temp_dif / time_dif_min    # change in temperature per minute
+                    humid_rate_min = humid_dif / time_dif_min  # change in humidity per minute
+            
+            print("Temperature rate of change: ", temp_rate_min)
+            print("Humidity rate of change: ", humid_rate_min)
+        
+            if temp_rate_min == None:  # no data generated if temp_rate_min is still None, date must be invalid
+                print("There are no data for the data.")
+                return
+
+            # if rate at which the temperature is changing exceeds threshold 
+            if temp_rate_min > self.args.tempRate or temp_rate_min < -self.args.tempRate:
+                self.create_seekpoint()
+                if self.cameraUuid == None:  # self.cameraUuid will be none if cameraName is invalid
+                    print("Invalid camera name, no seekpoint created.")
+                else:  
+                    print("Temperature exceeded threshold, seekpoint created.")
+
+            # if rate at which the humidity is changing exceeds threshold 
+            elif humid_rate_min > self.args.humidRate or humid_rate_min < -self.args.humidRate:
+                self.create_seekpoint()
+                if self.cameraUuid == None:  # self.cameraUuid will be none if cameraName is invalid
+                    print("Invalid camera name, no seekpoint created.")
+                else:
+                    print("Humidity exceeded threshold, seekpoint created.")
+            else:
+                print("No seekpoints created.")
+
+        else: # if the user chose the option of past 
+            running = True
+
+            temp_list = [] # list of temperature in Fahrenheit
+            humid_list = [] # list of humidities
+            time_list = [] # list of times in milliseconds 
+
+            while running == True:
+                print("Processing ... ")
+                end_time = int(round(time.time() * 1000))  # end time of data collection is always the current time
+                start_time = end_time - (1000 * 210)   # start time of data collection is always 210 seconds before current time
+
+                climate_data = self.climate_data(start_time, end_time)
```

Comment:
I got an error here (I think) where the connection was closed without response.  You should handle this gracefully and just keep going to the next loop